iidm/iidm-criteria/src/main/java/com/powsybl/iidm/criteria/SingleNominalVoltageCriterion.java

```diff
@@ -67,15 +68,34 @@ public VoltageInterval getVoltageInterval() {
     }
 
     public static class VoltageInterval {
-        Double nominalVoltageLowBound;
-        Double nominalVoltageHighBound;
-        Boolean lowClosed;
-        Boolean highClosed;
+        private static final VoltageInterval NULL_INTERVAL = new VoltageInterval();
 
-        public VoltageInterval() {
+        double nominalVoltageLowBound;
+        double nominalVoltageHighBound;
+        boolean lowClosed;
+        boolean highClosed;
+
+        private VoltageInterval() {
+            this.nominalVoltageLowBound = Double.NaN;
+            this.nominalVoltageHighBound = Double.NaN;
         }
 
-        public VoltageInterval(Double nominalVoltageLowBound, Double nominalVoltageHighBound, Boolean lowClosed, Boolean highClosed) {
+        public static VoltageInterval nullInterval() {
+            return NULL_INTERVAL;
+        }
+
+        public VoltageInterval(double nominalVoltageLowBound, double nominalVoltageHighBound, boolean lowClosed, boolean highClosed) {
```

Comment:
The problem here was that at least one of the bounds was not defined, the interval was considered as a "null interval"  and was simply ignored.
Thus, the following criterion was true for every network element!

```xml
 "nominalVoltageCriterion" : {
      "type" : "SINGLE_NOMINAL_VOLTAGE",
      "voltageInterval" : {
        "nominalVoltageHighBound" : 230.0,
        "highClosed" : true
      }
    }
```
I choose to stick with the existing principle that both bounds should be defined for the voltageInterval to be defined, but
I secured the class. Now, we can still have a null interval, but it must be created explicitly to prevent errors.

Reply:
Seen with @annetill and @EtienneLt: I will adapt this PR to have the same behaviour as for the duration interval. It will thus be possible to have an interval with a single bound. In the case of the previous comment's example, the criteria will return `true` for values in [0; 230] (0 is the min value for valid nominal voltages).